src/application/pokegear/phone/scripts/phone_scripts_prof_oak.c

```diff
@@ -0,0 +1,158 @@
+#include "global.h"
+
+#include "application/pokegear/phone/phone_internal.h"
+#include "msgdata/msg/msg_0666.h"
+
+#include "unk_02005D10.h"
+#include "unk_0205BB1C.h"
+
+u16 PhoneCall_GetScriptId_ProfOak(PokegearPhoneCallContext *ctx, PokegearPhoneCallState *state) {
+    Pokedex *pokedex;
+    u16 dexCountParam;
+
+    state->scriptType = 0;
+    pokedex = Save_Pokedex_Get(ctx->saveData);
+    if (state->isScriptedCall == 2) {
+        state->scriptType = 0;
+        return PHONE_SCRIPT_082;
+    }
+    if (state->isIncomingCall != 0) {
+        dexCountParam = Pokedex_CountNationalDexOwned(pokedex) / 50;
+        if (dexCountParam == 0) {
+            dexCountParam = 1;
+        } else if (dexCountParam > 9) {
+            dexCountParam = 9;
+        }
+        return PHONE_SCRIPT_068 + dexCountParam;
+    }
+    if (state->phoneBookEntry->mapId == ctx->playerMapSec) {
+        return PHONE_SCRIPT_068;
+    }
+    if (Save_VarsFlags_CheckFlagInArray(ctx->saveVarsFlags, FLAG_SYS_OAK_ACKNOWLEDGED_NATIONAL_DEX_COMPLETION)) {
+        return PHONE_SCRIPT_081;
+    }
+    if (!Pokedex_GetNatDexFlag(pokedex) && Save_VarsFlags_CheckFlagInArray(ctx->saveVarsFlags, FLAG_SYS_OAK_ACKNOWLEDGED_JOHTO_DEX_COMPLETION)) {
+        return PHONE_SCRIPT_080;
+    }
+    state->scriptType = 5;
+    return PHONE_SCRIPT_NONE;
+}
+
+BOOL GearPhoneCall_ProfOak(PokegearPhoneCallContext *ctx) {
+    PokegearPhoneCallState *state = &ctx->state;
+    Pokedex *pokedex;
+    int r6;
```

Comment:
Documentation:
Name local variables in function
Etc for file

Reply:
Leaving as future to-do